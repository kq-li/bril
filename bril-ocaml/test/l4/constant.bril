# ARGS: constant
@f() {
  y: int = const 0;
  ret y;
}

@main(arg: int) {
  x: int = const 1;
  y: int = const 5;
  z: int = const 6;
  cond: bool = const true;
  br cond .t .f;
.t:
  x: int = const 2;
  y: int = const 5;
  jmp .end;
.f:
  z: int = add x y;
  cond: bool = const false;
  jmp .end;
.end:
  x: int = const 7;
  z: int = call @f;
  ret;
}
