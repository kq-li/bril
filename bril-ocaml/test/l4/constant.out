current calculation: constant propagation
ins:
	block_0 -> {}
outs:
	block_0 -> {y: 0}
current calculation: constant propagation
ins:
	block_0 -> {}
	label_t -> {cond: true, x: 1, y: 5, z: 6}
	label_f -> {cond: true, x: 1, y: 5, z: 6}
	label_end -> {cond: ?, x: ?, y: 5, z: 6}
outs:
	block_0 -> {cond: true, x: 1, y: 5, z: 6}
	label_t -> {cond: true, x: 2, y: 5, z: 6}
	label_f -> {cond: false, x: 1, y: 5, z: 6}
	label_end -> {cond: ?, x: 7, y: 5, z: ?}
